test/env/run_e2e_tests.sh

```diff
@@ -6,6 +6,19 @@ BUILD_SCM_BRANCH=${BUILD_SCM_BRANCH:-"master"}
 BUILD_SCM_REVISION=${BUILD_SCM_REVISION:-"latest"}
 SKIP_TEST=${SKIP_TEST:-no}
 
+if [[ $BUILD_SCM_REVISION != "latest" ]] ; then
+    while : ; do
+        check_suites=$(curl -s -H "Accept: application/vnd.github.antiope-preview+json" https://api.github.com/repos/Juniper/contrail-operator/commits/$BUILD_SCM_REVISION/check-suites)
+        status=$(echo "$check_suites" | jq -cr '.check_suites[]|select(.app.slug == "google-cloud-build").status')
+        [[ $status != "completed" ]] && echo "Waiting for upstream job. Current status: $status" && sleep 120 && continue
```

Comment:
I'm not a fan of sleep - especially for 2 minutes in CI job that may have timeout, because it adds fixed 2 minutes to job's time.
Wouldn't it be possible to make it somehow use loop which has smaller value of sleep (e.g. 5 seconds) and periodically check for condition?

Maybe I'm not into what's going on here exactly and why, but this sleep stresses me :)

Reply:
I agree that we can try to poll this status more frequently :)

Reply:
Yeah I unfortunately this 2 minutes is not a coincident. As an unauthorised user I we issue 60 requests per hour https://developer.github.com/v3/#rate-limiting.

I may try to enable authentication if you would like to

Reply:
I will try to enable this authentication

Reply:
Great :D I know that storing credentials for authentication in CI is always a big challenge, so if authentication would generate security problems then I may accept these 2 minutes ;)

Reply:
Fixed I moved this code to Jenkinsfile and used token and I reduced sleep to 5

Reply:
I will apply this Jenkinsfile on Jenkins as soon as you approve